### PR TITLE
Fix DeprecationWarning

### PR DESCRIPTION
**Pull request by ericdasse-stonal**
`datetime.datetime.utcnow` has been deprecated by Python and will soon be removed by Python 3.14 (see warning text below)

Besides, this warning pops up everytime someone with a recent version of Python (3.12+) uses this plugin, which makes the testing experience uncomfortable. I thus thought this contribution would be nice

```
.venv/lib/python3.12/site-packages/pytest_postgresql/retry.py:22: DeprecationWarning: datetime.datetime.utcnow() is deprecated and scheduled for removal in a future version. Use timezone-aware objects to represent datetimes in UTC: datetime.datetime.now(datetime.UTC).
    time: datetime = datetime.utcnow()

-- Docs: https://docs.pytest.org/en/stable/how-to/capture-warnings.html
```

Chore that needs to be done:

* [x] Add newsfragment `pipenv run towncrier create [issue_number].[type].rst`

Types are defined in the pyproject.toml, issue_numer either from issue tracker or the Pull request number

### CODE REVIEW

File: pytest_postgresql/retry.py [hunk 1]
```diff
@@ -19,7 +19,7 @@ def retry(
     ... ::
         FATAL:  the database system is starting up
     """
-    time: datetime = datetime.utcnow()
+    time: datetime = datetime.now(UTC)
```
Comment (fizyk) [inline pytest_postgresql/retry.py:22]: This change is not compatible with python 3.8 (eol at October 2024), 3.9 (eol at October 2025) and 3.10 (eol at October 2026).
I'd want to have some compatibility layer for that.

**Review by fizyk — CHANGES_REQUESTED** [review]
And a newsfragment will be needed

File: pytest_postgresql/retry.py (hunk 1 above)
Comment (ericdasse-stonal) [reply pytest_postgresql/retry.py]: Done!

**Comment by ericdasse-stonal** [comment]
> @ericdasse-stonal could you add newsfragment with description as well?

Sure. To do that, do I need to create an issue first?

**Comment by fizyk** [comment]
No need, you can refer to this pr number as well

**Comment by ericdasse-stonal** [comment]
> No need, you can refer to this pr number as well

Alright. I took care of it

**Review by fizyk — APPROVED** [review]

**Pull request merged**